server/openapi_server/test/test_deidentified_notes_controller.py

```diff
@@ -2,36 +2,212 @@
     """DeidentifiedNotesController integration test stubs"""
 
-    def test_deidentified_notes_read_all(self):
-        """Test case for deidentified_notes_read_all
+    @patch('openapi_server.utils.annotator_client.get_annotations', new=mock_get_annotations)
+    def test_masking_char(self):
+        """Test case for de-identification with masking character
+        """
+        # Mask all fields, with different characters for each one
+        masking_char_request = {
+            "note": SAMPLE_NOTE,
+            "deidentificationConfigurations": [
+                {
+                    "deidentificationStrategy": {"maskingCharConfig": {"maskingChar": "*"}},
+                    "annotationTypes": ["text_physical_address"]
+                },
+                {
+                    "deidentificationStrategy": {"maskingCharConfig": {"maskingChar": "_"}},
+                    "annotationTypes": ["text_person_name"]
+                },
+                {
+                    "deidentificationStrategy": {"maskingCharConfig": {"maskingChar": "-"}},
+                    "annotationTypes": ["text_date"]
+                }
+            ]
+        }
+
+        response = self.client.open(
+            DEIDENTIFIER_ENDPOINT_URL,
+            method='POST',
+            headers={'Accept': 'application/json'},
+            data=json.dumps(masking_char_request),
+            content_type='application/json'
+        )
+        self.assertStatus(response, 201, 'Response body is : ' + response.data.decode('utf-8'))
+        response_data = response.json
+
+        # This is what the deidentified note *should* look like (based on how we know the annotators will annotate)
+        expected_deidentified_text = "____ __________ came back from ******* yesterday, -- -------- ----."
+        self.assertEqual(response_data['deidentifiedNote']['text'], expected_deidentified_text)
+
+        # Masking char de-identification doesn't change any annotation character addresses
+        self.assertEqual(response_data['deidentifiedAnnotations'], response_data['originalAnnotations'])
+
+    @patch('openapi_server.utils.annotator_client.get_annotations', new=mock_get_annotations)
+    def test_redact(self):
+        """Test case for de-identification with redaction
+        """
+        redact_request = {
+            "note": SAMPLE_NOTE,
+            "deidentificationConfigurations": [{
+                "deidentificationStrategy": {
+                    "redactConfig": {},
+                },
+                "annotationTypes": ["text_physical_address", "text_person_name", "text_date"]
+            }]
+        }
+
+        response = self.client.open(
+            DEIDENTIFIER_ENDPOINT_URL,
+            method='POST',
+            headers={'Accept': 'application/json'},
+            data=json.dumps(redact_request),
+            content_type='application/json'
+        )
+        self.assertStatus(response, 201, 'Response body is : ' + response.data.decode('utf-8'))
```

Comment:
Must be 200 once changed in the controller. First make sure the deidentifier specification states 200, if not open a PR to implement this schema change.